packages/@lwc/engine/src/framework/hooks.ts

```diff
@@ -41,36 +39,42 @@ function setElementShadowToken(elm: Element, token: string | undefined) {
 }
 
 export function updateNodeHook(oldVnode: VNode, vnode: VNode) {
-    const { text } = vnode;
+    const {
+        elm,
+        text,
+        owner: { renderer },
```

Comment:
in other places ([base-lightning-element](https://github.com/salesforce/lwc/pull/1879/files#diff-e81cd4c8dcd0d0a641f60cdfc8ef0dadR382)) i have seen a different approach, ex:

```js
const {
        elm,
        text,
        owner: { 
            renderer: { setText }
        },
} = vnode;
// ...
setText(elm, text!);
```

whats the preferred way of usage? when to use one and when to use another?

Reply:
It doesn't really matter because none of the methods rely on `this` but we can keep it consistent.